src/rewrite/steward.ts

```diff
@@ -454,3 +463,29 @@ export function genericUpdateTimeHeld(
   wildcard.owner = ownerString;
   wildcard.save();
 }
+
+export function handleCollectLoyalty(event: CollectLoyalty): void {
+  // Phase 1: reading and getting values.
+  let collectedLoyaltyTokens = event.params.amountRecieved;
+  let patronAddress = event.params.patron;
+  let tokenId = event.params.tokenId;
+  let patron = PatronNew.load(patronAddress.toHexString());
+  let numberOfTokensHeldByUserAtBeginningOfTx = BigInt.fromI32(
+    patron.tokens.length
+  );
+
+  // Phase 2: calculate new values.
+  let newCollectedLoyaltyTokens = patron.totalLoyaltyTokens.plus(
+    collectedLoyaltyTokens
+  );
+
+  let newTotalLoyaltyTokensIncludingUnRedeemed = patron.totalLoyaltyTokensIncludingUnRedeemed.plus(
```

Comment:
Reading this logic and just wondering if the newTotalLoyaltyTokensIncludingUnRedeemed will always be accurate.

Say i have 2 tokens, held each for 2 days, so when I run this one the one token, we should have 2 tokens x 2 days, therefore my newTotalLoyaltyTokensIncludingUnRedeemed =4 , and my actual collected loyalty tokens is 2. 

Now immediately, I collect the other 2 loyalty tokens on my other coin, now the logic gets run again and newTotalLoyaltyTokensIncludingUnRedeemed = 8, and my actual tokens collected is 4?

Let me know if this makes sense. I could definitely be missing something

Reply:
Yeah, it should be:
```
  let newTotalLoyaltyTokensIncludingUnRedeemed = patron.totalLoyaltyTokensIncludingUnRedeemed.plus(
    numberOfTokensHeldByUserAtBeginningOfTx.times(timeSinceLastUpdate.times(tokenGenerationRate))
  );
```


The `patron.tokens` is unfortunately incorrect on the `patronNew` entity.

And there is some funky stuff happening with the `patron.tokens` on the original (legacy) `patron` entity.

There is honestly tons of work required on the graph :see_no_evil:

Reply:
Fixing the previous tokens list is a blocker for this task. I'll look into it soon. Unless you have capacity @moose-code ?

This code should have a comment above it anyway, because it isn't super obvious and a bit hacky. The comment should say that this is a temporary fix until we change how loyalty tokens are generated.

Reply:
@JasoonS Is this still relevant after changes done from the weekend @moose-code

Reply:
I'll close this after the hackathon work.